packages/blade/src/components/Link/BaseLink/StyledBaseLink.web.tsx

```diff
@@ -1,33 +1,50 @@
+import React from 'react';
 import styled from 'styled-components';
 import getStyledLinkStyles from './getStyledLinkStyles';
 import type { StyledBaseLinkProps } from './types';
-import { castWebType, getIn, makeBorderSize, makeMotionTime } from '~utils';
+import { castWebType, getIn, getStyledProps, makeBorderSize, makeMotionTime } from '~utils';
+import { getBaseBoxStyles, getDependencyProps } from '~components/Box/BaseBox/getBaseBoxStyles';
 
 const StyledLink = styled.button.attrs<StyledBaseLinkProps>((props: StyledBaseLinkProps) => ({
   ...props.accessibilityProps,
-}))<StyledBaseLinkProps>((props) => ({
-  ...getStyledLinkStyles(props),
-  display: 'inline-block',
-  borderRadius: makeBorderSize(props.theme.border.radius.small),
-  transitionProperty: 'box-shadow',
-  transitionTimingFunction: getIn(props.theme.motion, props.motionEasing),
-  transitionDuration: castWebType(makeMotionTime(getIn(props.theme.motion, props.motionDuration))),
-  '.content-container': {
-    width: 'max-content',
+}))<StyledBaseLinkProps>((props) => {
+  // In other components, we wrap the component by div. But for link and text, we wanted to avoid that to not add unneccessary div around and cause layout any issues
+  // So we build the styled-props specifically for link and text
+  const styledPropsStyles = getStyledProps(props);
+  const styledPropsMemoDependency = getDependencyProps(styledPropsStyles);
+  const styledPropsCSSObject = React.useMemo(
+    () => getBaseBoxStyles({ ...styledPropsStyles, theme: props.theme }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [styledPropsMemoDependency],
+  );
```

Comment:
Whats happening here? 🤔

Reply:
https://github.com/razorpay/blade/pull/1016#discussion_r1118297636 🙈 let me create a common hook out of this.

Reply:
done. Created `useStyledProps` which wraps all this